elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -148,8 +151,8 @@ def __trace(method, path, params, body, url, response, json, took, duration)
         #
         # @api private
         def __raise_transport_error(response)
-          error = ERRORS[response.status] || ServerError
-          raise error.new "[#{response.status}] #{response.body}"
+          error = ERRORS[response.status].new(response) || ServerError.new(response)
```

Comment:
Hmm, I think the intention of the `ERRORS[response.status]` was to _try_ finding the class from the status code -- I don't think the new code will work when the status is not found in the `ERRORS` array, and the initialization will fail?